### PR TITLE
Add support for NewId package

### LINKED ISSUE #51: Add support for NewId
[NewId](https://github.com/phatboyg/NewId) is a sequential id generator analagous to sequential guids based on erlang's flake, and is the underlying id type i need to use for a few of my projects so having support would be ideal

### PR DESCRIPTION
**Pull request by Khitiara**
Fixes #51 
PR is in draft as I have yet to add tests

### CODE REVIEW

File: src/StronglyTypedIds.Attributes/StronglyTypedIdBackingType.cs [hunk 1]
```diff
@@ -17,5 +17,6 @@ public enum StronglyTypedIdBackingType
         String = 3,
         Long = 4,
         NullableString = 5,
+        NewId = 6,
```
Comment (andrewlock) [inline src/StronglyTypedIds.Attributes/StronglyTypedIdBackingType.cs:20]: I think maybe we should call this `MassTransitNewId`?

**Review by andrewlock — COMMENTED** [review]
I'm not familiar with NewId, but this looks good to me! If you could add some unit + integration tests I'd be happy to merge it, thanks! 🙂

**Comment by Khitiara** [comment]
I'll note the serialization routines are a bit more complicated than id hope to write integration tests for, as NewId is a zero-dependency library and does not serialize cleanly by default on its own
